### PR TITLE
Structure available_datasets and add tools overview

### PR DESCRIPTION
**Pull request by harterrt**
`available_datasets` is a good overview of all datasets, but it has two
large readability issues:
  * it lacks consistent structure between datasets
  * it lacks consistent style
`choosing_a_dataset` fixes these problems and adds a clear purpose to
the document.

`analyzing_telemetry_data` provides a quick, opinionated overview of the
major tools for analyzing telemetry data.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/telemetry-batch-view/136)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by harterrt** [comment]
@fbertsch r?

File: docs/choosing_a_dataset.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+## Mobile Metrics
+
+There are several tables owned by the mobile team documented
+[here](https://wiki.mozilla.org/Mobile/Metrics/Redash): 
+
+* android_events
+* android_clients
+* android_addons
+* and mobile_clients tables are documented here:
```
Comment (maurodoglio) [inline docs/choosing_a_dataset.md]: missing link?

**Comment by fbertsch** [comment]
Review status: 0 of 3 files reviewed at latest revision, 6 unresolved discussions.

---

*[docs/analyzing_telemetry_data.md, line 12 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_P0NDXFvUxiq4sNI2:-KV_P0NDXFvUxiq4sNI3:b-xnqhbp) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/analyzing_telemetry_data.md#L12)):*
> ```Markdown
> ## Table of Contents
> 
> * Ad Hoc Analysis
> ```

Should we include TMO (histograms + evolutions) on here? It would probably be better if people use that to get their information if they can, since it's less resource intensive and usually requires less handholding.

---

*[docs/analyzing_telemetry_data.md, line 39 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_PIuCmBbK0iugWAoN:-KV_PIuDANWqxx2zF5JP:b-469c70) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/analyzing_telemetry_data.md#L39)):*
> ```Markdown
> 
> Using an ATMO cluster allows you to perform more complicated analyses than are
> possible with SQL on STMO. You can create ATMO clusters with up to 20 worker
> ```

I'm pretty sure this has been upped to 30 nodes

---

*[docs/choosing_a_dataset.md, line 7 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_PZYD1TKats0-L8bN:-KV_PZYEBOyekxnuVh3B:b-goodx4) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L7)):*
> ```Markdown
> is the most useful for your analysis.
> 
> Before making final decisions from your findings, consider getting your query
> ```

We should have some sort of hard requirement here, but that's probably a larger discussion

---

*[docs/choosing_a_dataset.md, line 66 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_QM7wZItQmCg2UTrD:-KV_QM7xdGRCtzeLdj3V:b-1qlcex) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L66)):*
> ```Markdown
> ### Rows and Columns
> Each row in the `longitudinal` dataset represents one `client_id`, which is a
> rough approximation to a user.  Each column represents a field from the main
> ```

rough approximation "of" as user (probably)

---

*[docs/choosing_a_dataset.md, line 126 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_RHVzfX1m3s-vytIh:-KV_RHVzfX1m3s-vytIi:b830veu) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L126)):*
> ```Markdown
> Each row represents one combinations of the factor columns.
> 
> ### Background and Caveats
> ```

background: Why use HLLs, instead of direct counts? (Users might be confused here).

---

*[docs/choosing_a_dataset.md, line 137 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_RjW3bg7XvomcdGJR:-KV_RjW3bg7XvomcdGJS:bsdziad) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L137)):*
> ```Markdown
> 
> ### Rows and Columns
> Example dimensions include channel and country, example statistics include
> ```

Maybe more direct here: "Columns are the set of dimensions, and the crash statistics". Each row is a distinct set of dimensions, along with their associated crash stats. (I'm not actually sure about rows being unique in dimensions).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136)*

**Comment by harterrt** [comment]
Review status: 0 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[docs/analyzing_telemetry_data.md, line 12 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_P0NDXFvUxiq4sNI2:-KV_vLxk-bSg80me2soH:blzt76m) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/analyzing_telemetry_data.md#L12)):*
<details><summary><i>Previously, fbertsch (Frank Bertsch) wrote…</i></summary>
> Should we include TMO (histograms + evolutions) on here? It would probably be better if people use that to get their information if they can, since it's less resource intensive and usually requires less handholding.
</details>
I'd like to open a bug and add this in a future revision. I haven't done any work with TMO, is it used frequently by our clients?

---

*[docs/analyzing_telemetry_data.md, line 39 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_PIuCmBbK0iugWAoN:-KV_s75yE_1x9wp4BWf3:bo6djp1) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/analyzing_telemetry_data.md#L39)):*
<details><summary><i>Previously, fbertsch (Frank Bertsch) wrote…</i></summary>
> I'm pretty sure this has been upped to 30 nodes
</details>
Just checked, it's still at 20 for me.

---

*[docs/choosing_a_dataset.md, line 7 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_PZYD1TKats0-L8bN:-KV_v44vtLRkRNX8pHgA:b2nhb2c) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L7)):*
<details><summary><i>Previously, fbertsch (Frank Bertsch) wrote…</i></summary>
> We should have some sort of hard requirement here, but that's probably a larger discussion
</details>
Ack

---

*[docs/choosing_a_dataset.md, line 66 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_QM7wZItQmCg2UTrD:-KV_sYi-Q0tCzp3UT0Ax:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L66)):*
<details><summary><i>Previously, fbertsch (Frank Bertsch) wrote…</i></summary>
> rough approximation "of" as user (probably)
</details>
Done.

---

*[docs/choosing_a_dataset.md, line 137 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_RjW3bg7XvomcdGJR:-KV_tUhLLduUXugbrIWW:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L137)):*
<details><summary><i>Previously, fbertsch (Frank Bertsch) wrote…</i></summary>
> Maybe more direct here: "Columns are the set of dimensions, and the crash statistics". Each row is a distinct set of dimensions, along with their associated crash stats. (I'm not actually sure about rows being unique in dimensions).
</details>
Done.

---

*[docs/choosing_a_dataset.md, line 151 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_MAtlr1xBfpnGFOK_-r1-151:-KV_sj1Ux6eA6OOqFFRV:bx0cw3k) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L151)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> missing link?
</details>
Done. This is actually linked above.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136)*

**Comment by fbertsch** [comment]
Review status: 0 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[docs/analyzing_telemetry_data.md, line 12 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_P0NDXFvUxiq4sNI2:-KVa4P1CiK4uwqN5Gt_g:b-q9rjep) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/analyzing_telemetry_data.md#L12)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> I'd like to open a bug and add this in a future revision. I haven't done any work with TMO, is it used frequently by our clients?
</details>
TMO is probably used *more* than ATMO or STMO, check out the results from our survey [0] (with plenty of caveats, obviously). Mostly engineers, who go "how is this histogram behaving?", and then when TMO doesn't give them enough, they move on to *TMO. (I'm pretty sure this is the process you will be trying to capture). Note, however, that it's different for managers/POs.

However, I do agree that a separate bug makes sense. If you're not comfortable with TMO I can try to find time this quarter to document it.

[0] https://people-mozilla.org/~fbertsch/tmo_survey_analysis.html

---

*[docs/analyzing_telemetry_data.md, line 39 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_PIuCmBbK0iugWAoN:-KVa5LIKfH1vLO23YQ-5:b-6hkrdg) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/analyzing_telemetry_data.md#L39)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> Just checked, it's still at 20 for me.
</details>
Consider me corrected :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136)*

**Comment by harterrt** [comment]
Review status: 0 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[docs/analyzing_telemetry_data.md, line 12 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_P0NDXFvUxiq4sNI2:-KVef53hpzFaYGxeUhlM:b-71xsi1) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/analyzing_telemetry_data.md#L12)):*
<details><summary><i>Previously, fbertsch (Frank Bertsch) wrote…</i></summary>
> TMO is probably used *more* than ATMO or STMO, check out the results from our survey [0] (with plenty of caveats, obviously). Mostly engineers, who go "how is this histogram behaving?", and then when TMO doesn't give them enough, they move on to *TMO. (I'm pretty sure this is the process you will be trying to capture). Note, however, that it's different for managers/POs.
> 
> However, I do agree that a separate bug makes sense. If you're not comfortable with TMO I can try to find time this quarter to document it.
> 
> [0] https://people-mozilla.org/~fbertsch/tmo_survey_analysis.html
</details>
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1314973

---

*[docs/analyzing_telemetry_data.md, line 39 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_PIuCmBbK0iugWAoN:-KVeeMJjsQEn_UOIY9yP:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/analyzing_telemetry_data.md#L39)):*
<details><summary><i>Previously, fbertsch (Frank Bertsch) wrote…</i></summary>
> Consider me corrected :)
</details>
Done.

---

*[docs/choosing_a_dataset.md, line 126 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_RHVzfX1m3s-vytIh:-KVeeWnEugj3yhhzr5At:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L126)):*
<details><summary><i>Previously, fbertsch (Frank Bertsch) wrote…</i></summary>
> background: Why use HLLs, instead of direct counts? (Users might be confused here).
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136)*

**Comment by fbertsch** [comment]
Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136)*

**Comment by fbertsch** [comment]
@harterrt :shipit:

**Comment by harterrt** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[docs/choosing_a_dataset.md, line 151 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136#-KV_MAtlr1xBfpnGFOK_-r1-151:-KVfho4YtBRDWMUgKEJ1:bd76o5m) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d7ec0981f951e3f17f746d81451ef6dc52e4b0e7/docs/choosing_a_dataset.md#L151)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> Done. This is actually linked above.
</details>
@maurodoglio - LG?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/136)*

**Pull request merged**